File: 007)StoringData/app/src/main/java/com/example/a007storingdata/MainActivity.java
package com.example.a007storingdata;

import android.content.Context;
import android.content.SharedPreferences;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    private EditText editTextAge;
    private TextView textView;
    SharedPreferences sharedPreferences;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        editTextAge = findViewById(R.id.editTextAge);
        textView = findViewById(R.id.textView);

        sharedPreferences = this.getSharedPreferences("com.example.a007storingdata", Context.MODE_PRIVATE);
        int storedAge = sharedPreferences.getInt("storedAge", 0);
        textView.setText("Your Age: " + storedAge);
    }

    public void save(View view){
        if (!editTextAge.getText().toString().matches("")){
            int age = Integer.parseInt(editTextAge.getText().toString());
            textView.setText("Your Age: " + age);

            sharedPreferences.edit().putInt("storedAge",age).apply();
        }
    }

    public void delete(View view){
        int storedData = sharedPreferences.getInt("storedAge", 0);
        if (storedData != 0){
            sharedPreferences.edit().remove("storedAge").apply();
            textView.setText("Your Age: 0");
        }
    }
}

File: 048)StoringData/app/src/main/java/com/elyesasimsek/storingdata/MainActivity.kt
package com.elyesasimsek.storingdata

import android.content.SharedPreferences
import android.os.Bundle
import android.view.View
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.databinding.DataBindingUtil
import com.elyesasimsek.storingdata.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding
    private lateinit var sharedPref: SharedPreferences
    private var ageFromPref: Int? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        //shared prefences - XML - Key-value
        sharedPref = this.getSharedPreferences("com.elyesasimsek.storingdata", MODE_PRIVATE)
        ageFromPref = sharedPref.getInt("age", -1)
        if (ageFromPref == -1){
            binding.textView.text = "Your age: "
        }else{
            binding.textView.text = "Your age: ${ageFromPref}"
        }
    }

    fun save(view: View){
        val myAge  = binding.editText.text.toString().toIntOrNull()
        if (myAge != null){
            binding.textView.text = "Your age: ${myAge}"
            sharedPref.edit().putInt("age", myAge).apply()
        }

    }

    fun delete(view: View){
        ageFromPref = sharedPref.getInt("age", -1)
        if (ageFromPref != -1){
            sharedPref.edit().remove("age").apply()
            binding.textView.text = "Your age: "
        }
    }
}